docs/spec_draft.md

```diff
@@ -341,6 +351,168 @@ only exists to establish data dependencies from `result` to `inputs`.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.all_to_all
+
+### Semantics
+
+Scatters `split_count` blocks of data, by splitting `operand` along
+`split_dimension`, from all the execution instances to all the execution
+instances. Each execution instance concatenates the received blocks along the
+`concat_dimension`.
+
+The following diagram shows how the scattered blocks of `operand`, from all the
+execution instances, are concatenated to form `result` in each execution
+instance.
+
+<img align="center" src="spec_draft/all_to_all.svg" />
```

Comment:
Check out the internal deck for the proposed update to the diagram.

Feedback:
  1. Switched to the proposed new terminology.
  2. Added more explanations next to `operand` and especially next to `result`.
  3. Switched the color scheme to match the colors from gather/scatter.
  4. Reduced the number of colors from 4 to 2 to emphasize which blocks end up in which process.

Reply:
Thanks!